### PR TITLE
[Bug](pipeline) do not treat BLOCKED->FINISHED as error when wake_up_early is true

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
This pull request improves the state transition logic for `PipelineTask` to handle rare race conditions when a delayed `wake_up()` call arrives after a task has already finished or finalized. It introduces an extended state transition table for the `_wake_up_early` scenario, ensures that illegal or backward state transitions are prevented, and adds comprehensive unit tests for these new behaviors.

**State transition logic improvements:**

* Added a new `WAKE_UP_EARLY_LEGAL_STATE_TRANSITION` table to allow certain additional state transitions when `_wake_up_early` is set, specifically permitting `BLOCKED→FINISHED`, `FINISHED→RUNNABLE`, and `FINALIZED→RUNNABLE` transitions under this mode.
* Updated `_state_transition` to use the appropriate transition table based on `_wake_up_early`, and to treat `FINISHED/FINALIZED→RUNNABLE` as a legal but no-op transition (state remains unchanged).

**Handling of delayed wake-up calls:**

* Modified `wake_up()` to avoid resubmitting a task to the scheduler if a delayed wake-up arrives after the task has already finished or finalized, preventing unintended re-execution.

**Documentation and testing:**

* Expanded the state machine documentation in `PipelineTask` to describe the new transitions allowed by `_wake_up_early`.
* Added thorough unit tests to verify both normal and `_wake_up_early` state transitions, including edge cases for no-op transitions and correct handling of delayed wake-ups.

### CODE REVIEW

**Review by Gabriel39 — DISMISSED** [review]

File: be/src/exec/pipeline/pipeline_task.cpp [hunk 1]
```diff
@@ -1063,11 +1063,19 @@ Status PipelineTask::_state_transition(State new_state) {
     }
     _task_profile->add_info_string("TaskState", _to_string(new_state));
     _task_profile->add_info_string("BlockedByDependency", _blocked_dep ? _blocked_dep->name() : "");
-    if (!LEGAL_STATE_TRANSITION[(int)new_state].contains(_exec_state)) {
+    const auto& table =
+            _wake_up_early ? WAKE_UP_EARLY_LEGAL_STATE_TRANSITION : LEGAL_STATE_TRANSITION;
```
Comment (HappenLee) [inline be/src/exec/pipeline/pipeline_task.cpp:1067]: 这块有几组啊？能不能后面搞成一个二维数组

File: be/src/exec/pipeline/pipeline_task.cpp [hunk 2]
```diff
@@ -1063,11 +1063,19 @@ Status PipelineTask::_state_transition(State new_state) {
...
         return Status::InternalError(
                 "Task state transition from {} to {} is not allowed! Task info: {}",
                 _to_string(_exec_state), _to_string(new_state), debug_string());
     }
+    // FINISHED/FINALIZED → RUNNABLE is legal under wake_up_early (delayed wake_up() arriving
+    // after the task already terminated), but we must not actually move the state backwards.
+    if ((_exec_state == State::FINISHED || _exec_state == State::FINALIZED) &&
+        new_state == State::RUNNABLE) {
+        return Status::OK();
+    }
```
Comment (HappenLee) [inline be/src/exec/pipeline/pipeline_task.cpp:1079]: 这里应该加个else，等于统一用最后的return来返回啊，代码格式得讲究一下

File: be/src/exec/pipeline/pipeline_task.cpp (hunk 1 above)
Comment (BiteTheDDDDt) [reply be/src/exec/pipeline/pipeline_task.cpp:1067]: 就两个

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
